portfolio/src/main/java/com/google/sps/servlets/LogInServlet.java

```diff
@@ -24,11 +24,20 @@
 @WebServlet("/login")
 public class LogInServlet extends HttpServlet {
   
+  final String ROOT_URL = "/";
+  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("application/json;");
     UserService userService = UserServiceFactory.getUserService();
-    boolean loggedIn = userService.isUserLoggedIn();
-    response.getWriter().println(new Gson().toJson(loggedIn));
+    List<String> loginInfo = new ArrayList<String>();
+    String loggedIn = Boolean.toString(userService.isUserLoggedIn());
+    loginInfo.add(loggedIn);
+    if (loggedIn.equals("false")) {
```

Comment:
You should put the boolean value in a variable and use that in the if statement. Checking the value of a string boolean value is very inefficient and harder to read.